service/src/main/kotlin/app/cash/backfila/service/runner/statemachine/VariableCapacityChannel.kt

```diff
@@ -48,17 +49,20 @@ class VariableCapacityChannel<T>(
             if (buffer.size > 0) {
```

Comment:
I wonder if @keeferrourke might have an opinion since he is more familiar with coroutines. But he is away for a week. Seems reasonable to me though.